src/main/php/de/thekid/dialog/import/UploadMedia.php

```diff
@@ -0,0 +1,37 @@
+<?php namespace de\thekid\dialog\import;
+
+use de\thekid\dialog\processing\Processing;
+use io\File;
+use webservices\rest\{Endpoint, RestUpload};
+
+class UploadMedia extends Task {
+
+  public function __construct(private string $slug, private File $source, private Processing $processing) { }
+
+  public function execute(Endpoint $api) {
+    $resource= $api->resource('entries/{0}/images/{1}', [$this->slug, $this->source->filename]);
+    $transfer= [];
+
+    // Process targets first...
+    foreach ($this->processing->targets($this->source) as $kind => $target) {
+      $transfer[$kind]= $target;
```

Comment:
This used to include the following:

```php
/** Executes a given external command and returns its exit code */
private function execute(string $command, array<string> $args, $redirect= null): void {
  $p= new Process($command, $args, null, null, [STDIN, $redirect ?? STDOUT, STDERR]);
  if (0 === ($r= $p->close())) return;

  throw new IllegalStateException($p->getCommandLine().' exited with exit code '.$r);
}

// FIXME: Uploading files that take longer than ~30 seconds is, for some reason,
// broken, and will result in a) the import tool crashing and b) the server ending
// up in an endless blocking loop. Use `curl` for videos instead.
if ('video' === $kind) {
  $this->execute('curl', [
    '-#',
    '-X', 'PUT',
    '-H', 'Authorization: '.$this->api->headers()['Authorization'],
    '-F', $kind.'=@'.strtr($target->getURI(), [DIRECTORY_SEPARATOR => '/']),
    $resource->uri(),
  ], ['null']);
} else {
  $transfer[$kind]= $target;
}
```

...which we might need to reinstate after integration-testing this PR.